saleor/plugins/vatlayer/__init__.py

```diff
@@ -89,3 +98,131 @@ def get_taxed_shipping_price(shipping_price, taxes):
     if not charge_taxes_on_shipping():
         taxes = None
     return apply_tax_to_price(taxes, DEFAULT_TAX_RATE_NAME, shipping_price)
+
+
+def apply_checkout_discount_on_checkout_line(
+    checkout_info: "CheckoutInfo",
+    lines: List["CheckoutLineInfo"],
+    checkout_line_info: "CheckoutLineInfo",
+    discounts: Iterable["DiscountInfo"],
+    line_price: Money,
+):
+    """Calculate the checkout line price with discounts.
+
+    Include the entire order voucher discount.
+    """
+    voucher = checkout_info.voucher
+    if not voucher or voucher.type == VoucherType.SHIPPING:
```

Comment:
Is it okay that we will take into account a voucher of type `SPECIFIC_PRODUCTS`? ☝️